src/audio/codec_adapter/codec_adapter.c

```diff
@@ -178,6 +179,7 @@ int codec_adapter_prepare(struct comp_dev *dev)
 	 * large enough to save the produced output samples.
 	 */
 	buff_size = MAX(mod->period_bytes, md->mpd.out_buff_size) * buff_periods;
```

Comment:
this calculation seems weird, on L167-169 we are dividing this value by period bytes to get buff_periods. So this relationship is roughly (accounting for integer division)
`md->mpd.out_buff_size ~ buff_periods * mod->period_bytes`

So shouldn't it be `MAX(mod->period_bytes * buff_periods, md->mpd.out_buff_size) ;`?

Or is this something wonky based on L171-173's math?

Reply:
@cujomalainey this calculation is definitely fishy but its not a trivial fix either. I will have to revisit this at some point later.

Reply:
Please make a bug so it is tracked, thanks

Reply:
@cujomalainey Im not sure if this is even a bug as of now. But I'll discuss this with @dbaluta